keytree/keytree.md

```diff
@@ -0,0 +1,146 @@
+# Key tree
+
+This is a _key blinding scheme_ for deriving hierarchies of public keys.
+
+The most important feature of this scheme is that a set of _public_ keys can be derived from a _public_ key,
+without the use of private keys. This allows a piece of software to generate unique receiving addresses
+without having the private key material available (e.g., an online merchant may keep only public keys on the server and generate invoices with unique keys without compromising security of the private keys).
+
+## Definitions
+
+### Scalar
+
+A _scalar_ is an integer modulo [Ristretto group](https://ristretto.group) order `|G| = 2^252 + 27742317777372353535851937790883648493`.
+
+Scalars are encoded as 32-byte strings using little-endian convention.
+
+Every scalar is required to be in a canonical (reduced) form.
+
+### Point
+
+A _point_ is an element in the [Ristretto group](https://ristretto.group).
+
+Points are encoded as 32-byte [data types](#data-type) in _compressed Ristretto form_.
+
+
+### Base point
+
+Ristretto base point in compressed form:
+
+```
+B = e2f2ae0a6abc4e71a884a961c500515f58e30b6aa582dd8db6a65945e08d2d76
+```
+
+### Derivation key
+
+Uniformly random secret 32-byte string used to derive blinding factors.
+
+### Xprv
+
+Stands for _extended private key_: consists of a secret [scalar](#scalar) and a [derivation key](#derivation-key).
+
+```rust
+struct Xprv {
+  scalar: Scalar,
+  dk: [u8; 32]
+}
+```
+
+### Xpub
```

Comment:
Could we add a note here that `Xpub` is not meant to be sent to anyone, and that we send the blinded representation as a point ? (either in this section or a different one, if that seems like a better fit!)

Reply:
actually, it is supposed to be sent to _someone_ who needs to track balances or create invoices, but not published to _everyone_. Will clarify!

Reply:
ah, even better clarification! thanks :) :)

Reply:
clarified: 
https://github.com/interstellar/slingshot/pull/87/commits/b5fe7049254993a6598fb71d1e62bffe0e5a344f